### ISSUE TITLE
Blank window when app starts automatically on login

### ISSUE THREAD

**Issue by kikijiki** [mattermost/desktop#713]
I confirm (by marking "x" in the [ ] below: [x]):

- [x] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [x] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [x] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

---

**Summary**

When logging in, Mattermost starts correctly but the window stays blank.
The page contents seem to be there (if I move the mouse I can see it changing when hovering on a link for example), but they are not rendered (same for the devtools window).
Closing and reopening the app fixes this (just closing the window is not enough).
This started happening with `v4.0.0`.

Things I tried:
- Reload/Clear cache and reload
- Uninstall, remove all mattermost folders from appdata and reinstall
- Update the graphics driver to the latest version

**Steps to reproduce**

Login, check Mattermost (happens every time).

- Operating System: Windows 7 64bit
- Mattermost Desktop App version: `4.0.0`
- Mattermost Server version: `4.6.0`
- GPU: NVIDIA GeForce GT 440
- GPU driver version: 390.77

**Comment by yuya-oc** [comment]
Probably there are some trouble in rendering though, unfortunately I'm not sure in detail. Would you try disabling GPU as a workaround?

`mattermost.exe -- --disable-gpu`

**Comment by kaksikanaa** [comment]
We have this same problem (on Win 7 PC's) when Mattermost starts at login but the needed VPN connection is established after it OR when PC wakes from the hibernation and again the Mattermost desktop app wakes before the VPN connection.  Previous version of the Mattermost app did not have this issue or it just was able to try to connect again a bit later on one or more times.

**Comment by yuya-oc** [comment]
Sorry, I misunderstood the situation. However not reproduced on my PC even when I use "Start app on login".

- Windows 10 Pro 64bit
- Mattermost Server version: 4.7.1
- GPU: NVIDIA GeForce GTX 660
- GPU driver version: 390.77

@jasonblais Possibly we need "Disable GPU acceleration" option. In startup, applying our workaround `--disable-gpu` is a little hard because command prompt is not available. An user needs to manually edit registry.

@kaksikanaa Is the window completely blank? As far as reading your report, I think that the app merely failed to connect to the server due to unestablished VPN. The order of startup is managed by Windows, so unfortunately there are no way we can do in source code. As a possible workaround, how about creating a batch file to wait VPN then launch the app?

**Comment by kikijiki** [comment]
@yuya-oc Adding `--disable-gpu` fixed the problem for me.

```
[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Run]
"Mattermost"="\"C:\\Users\\<username>\\AppData\\Local\\mattermost\\update.exe\" --processStart \"Mattermost.exe\" --process-start-args \"--hidden\""
↓
"Mattermost"="\"C:\\Users\\<username>\\AppData\\Local\\mattermost\\update.exe\" --processStart \"Mattermost.exe\" --process-start-args \"--hidden --disable-gpu\""
```

**Comment by kaksikanaa** [comment]
Great, he disable-gpu option resolved our case also, thanks! 

Would it be nice if user could find these option right in his/her settings/profile...?

**Comment by kikijiki** [comment]
Looks like a bug in electron, maybe related to electron/electron/issues/4380 or electron/electron/issues/8505 ?

**Comment by jasonblais** [comment]
@yuya-oc If we added the setting, would the steps be to first quit, then reopen the app, go to the Settings page, and disable GPU acceleration?

You brought it up before too, so maybe it's worth adding the setting to fix these issues. We can consider naming the setting so that it's clear to end users, perhaps something like

```
Title: Use GPU hardware acceleration
Help text: Disable this setting if you see a blank page after logging in to Mattermost. If enabled, the Mattermost UI is rendered more efficiently but can cause stability issues for some systems. Setting takes affect after restarting the app.
```

PS: Feedback welcome on settings text @kikijiki @kaksikanaa

**Comment by kikijiki** [comment]
Looks good to me :+1:

**Comment by kaksikanaa** [comment]
Thanks, the text is okay! Could this option be disabled as default?

**Comment by yuya-oc** [comment]
@jasonblais Yes if GPU acceleration is enabled by default.

"Disabled by default" is safer, but many users might not notice it's disabled even if it's applicable.

**Comment by jasonblais** [comment]
Thanks all!

Wondering how much benefit there is to have the option of enabling GPU acceleration if it's disabled by default. Does UI rendering performance degrade noticeably, or not really for the Mattermost app?

**Comment by avasconcelos114** [comment]
Actually, we've been having issues among our users too, and what makes things more difficult is that because of some security specifications, the Mattermost desktop app is ran through a separate program (written in C#). We've tried running it with the `--disable-gpu` option through that program to no avail. 

Having a hardware acceleration option would work wonders for us (especially if handled by the config.json file)

**Comment by Austin1** [comment]
I'm having same issue in Windows 7, Lenovo X260 (integrated graphics Intel HD 520). Almost always occurs on first load after fresh reboot. I close app using right click menu in taskbar and then reopen. On reopen I see the loading graphics and it works.

**Comment by yuya-oc** [comment]
I added the PR #734. I'm wondering if you could test whether your problem are gone with latest artifacts. 

Artifacts are available here. https://circleci.com/gh/yuya-oc/desktop/696#artifacts (if you can't see them, please try again later)

**Comment by yuya-oc** [comment]
Closed due to #734
